src/toolkit/matl_buy_policy.cc

```diff
@@ -199,10 +226,32 @@ void MatlBuyPolicy::AcceptMatlTrades(
   for (it = resps.begin(); it != resps.end(); ++it) {
     rsrc_commods_[it->second] = it->first.request->commodity();
     LGH(INFO3) << "got " << it->second->quantity() << " kg of "
-               << it->first.request->commodity();
+               << it->first.request->commodity()  << std::endl;
     buf_->Push(it->second);
   }
 }
 
+void MatlBuyPolicy::SetNextActiveTime() {
+  next_active_end_ = active_dist_->sample() + manager()->context()->time();
+  return;
+};
+
+void MatlBuyPolicy::SetNextDormantTime() {
+  if (next_dormant_end_ < 0) {}
+  else {
+    next_dormant_end_ = dormant_dist_->sample() + manager()->context()->time();
```

Comment:
doesn't this need to be
```suggestion
    next_dormant_end_ = dormant_dist_->sample() + next_active_end_;
```

Reply:
I think these are functionally equivalent? `SetNextDormantTime` is only called at the end of an active period, so `manager()->context()->time()` is the same as `next_active_end_`. If you think using `next_active_end_` is better, 👍